Barotrauma/BarotraumaClient/ClientSource/Screens/SubEditorScreen.cs

```diff
@@ -230,7 +230,22 @@ private static string GetSubDescription()
 
         private static LocalizedString GetTotalHullVolume()
         {
-            return $"{TextManager.Get("TotalHullVolume")}:\n{Hull.HullList.Sum(h => h.Volume)}";
+            float totalVolume = 0.0f;
+            float ballastVolume = 0.0f;
+            Hull.HullList.ForEach(h =>
+            {
+                totalVolume += h.Volume;
+
+                if (h.IsBallast())
+                {
+                    ballastVolume += h.Volume;
+                }
+            });
+
+            string retVal = $"{TextManager.Get("TotalHullVolume")}:\n{totalVolume}";
+            retVal += $" ({(ballastVolume / totalVolume * 100).ToString("0.00")} % {TextManager.Get("roomname.Ballast")})";
```

Comment:
I noticed that if `totalVolume` is `0.0f`, the division here could result in `Infinity`, which might cause unexpected behavior when formatting the string.

Reply:
I had the same thought when writing this, though `GetTotalHullVolume` is only called when at least one hull is selected so this should never happen.
That being said, I have no problem adding a defensive guard just in case.